src/Juvix/Compiler/Internal/Language.hs

```diff
@@ -170,9 +170,9 @@ data Literal
   = LitString Text
   | -- | `LitNumeric` represents a numeric literal of undetermined type
     LitNumeric Integer
-  | -- | `LitInteger` represents a literal of type `Int`
+  | -- | `LitInteger` represents a literal with trait `Integral`
     LitInteger Integer
-  | -- | `LitNatural` represents a literal of type `Nat`
+  | -- | `LitNatural` represents a literal with trait `Natural`
     LitNatural Integer
```

Comment:
The way that literals work in type checking is here:

https://github.com/anoma/juvix/blob/main/src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/TypeChecking/CheckerNew.hs#L1111-L1116

For a non-negative literal `n` the typechecker runs type inference on the literal wrapped in `fromNat` from the `FromNatural` trait:

```
fromNat {_} {{_}} `n`
```

The `Byte` type is then obtained via instance resolution.

This is why I updated the documentation for `LitInteger` and `LitNatural` to refer to traits rather than to types.